src/v/cloud_storage/remote_partition.h

```diff
@@ -132,6 +132,7 @@ class remote_partition
     static ss::future<erase_result> erase(
       cloud_storage::remote&,
       cloud_storage_clients::bucket_name,
+      const remote_path_provider& path_provider,
```

Comment:
You can store `remote_path_provider` inside the `materialized_resources` instance. It is available through `remote`. So you won't have to pass it directly into both archiver and remote_partition.

Reply:
I don't think we can since the path provider is tied to each topic since for example a recovery topic will have a different base label